modules/light-clients/08-wasm/types/misbehaviour_handle_test.go

```diff
@@ -469,372 +483,131 @@ package types_test
 //				}
 //			}, false,
 //		},
-//	}
-//
-//	for _, tc := range testCases {
-//		suite.Run(tc.name, func() {
-//			suite.SetupWasmTendermint()
-//			path = ibctesting.NewPath(suite.chainA, suite.chainB)
-//
-//			err := path.EndpointA.CreateClient()
-//			suite.Require().NoError(err)
-//
-//			tc.malleate()
-//
-//			clientState := path.EndpointA.GetClientState()
-//			clientStore := suite.chainA.App.GetIBCKeeper().ClientKeeper.ClientStore(suite.chainA.GetContext(), path.EndpointA.ClientID)
-//
-//			err = clientState.VerifyClientMessage(suite.chainA.GetContext(), suite.chainA.App.AppCodec(), clientStore, misbehaviour)
-//
-//			if tc.expPass {
-//				suite.Require().NoError(err)
-//			} else {
-//				suite.Require().Error(err)
-//			}
-//		})
-//	}
+// 	}
+
+// 	for _, tc := range testCases {
+// 		suite.Run(tc.name, func() {
+// 			// reset suite to create fresh application state
+// 			suite.SetupWasmWithMockVM()
+
+// 			endpoint := wasmtesting.NewWasmEndpoint(suite.chainA)
+// 			err := endpoint.CreateClient()
+// 			suite.Require().NoError(err)
+
+// 			tc.malleate()
+
+// 			clientStore := suite.chainA.App.GetIBCKeeper().ClientKeeper.ClientStore(suite.chainA.GetContext(), endpoint.ClientID)
+// 			clientState := endpoint.GetClientState()
+// 			foundMisbehaviour := clientState.CheckForMisbehaviour(
+// 				suite.chainA.GetContext(),
+// 				suite.chainA.App.AppCodec(),
+// 				clientStore, // pass in clientID prefixed clientStore
+// 				clientMessage,
+// 			)
+
+// 			suite.Require().Equal(tc.foundMisbehaviour, foundMisbehaviour)
+// 		})
+// 	}
 // }
 
-/* func (suite *TypesTestSuite) TestCheckForMisbehaviourGrandpa() {
+func (suite *TypesTestSuite) TestCheckForMisbehaviourTendermint() {
```

Comment:
Change name to `TestCheckForMisbehaviour`

Reply:
I thought about this too actually :) but the vm functions we have are specific to the Tendermint ClientState so I opted to leave it like this. happy to change it if other people have strong opinions!

Reply:
Sure, I will add here for the record that I have a strong opinion XD